### PR TITLE
CI against Ruby 2.6.0

### PR DESCRIPTION
**Pull request by koic**
Ruby 2.6.0 has been released and this Ruby version is available on CircleCI.

- https://www.ruby-lang.org/en/news/2018/12/25/ruby-2-6-0-released/
- https://hub.docker.com/r/circleci/ruby/tags/

```console
% docker pull circleci/ruby:2.6
2.6: Pulling from circleci/ruby
(snip)
```

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [ ] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [ ] Added tests.
* [ ] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](../blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [x] Run `bundle exec rake default`. It executes all tests and RuboCop for itself, and generates the documentation.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -235,13 +258,13 @@ jobs:
       - run:
           name: Upload coverage results to Code Climate
           command: |
-            ./tmp/cc-test-reporter sum-coverage tmp/codeclimate.*.json --parts 5 --output tmp/codeclimate.total.json
+            ./tmp/cc-test-reporter sum-coverage tmp/codeclimate.*.json --parts 6 --output tmp/codeclimate.total.json
```
[the remark this answers is not in the document]
Comment (koic) [reply .circleci/config.yml]: This change was adapted to the behavior when using Ruby 2.6.0.

```console
$ #!/bin/bash -eo pipefail
./tmp/cc-test-reporter sum-coverage tmp/codeclimate.*.json --parts 5 --output tmp/codeclimate.total.json
./tmp/cc-test-reporter upload-coverage --input tmp/codeclimate.total.json
Error: expected 5 parts, received 6 parts
Usage:
  cc-test-reporter sum-coverage [flags]

Flags:
  -o, --output string   output path (default "coverage/codeclimate.json")
  -p, --parts int       total number of parts to sum

Global Flags:
  -d, --debug   run in debug mode

Exited with code 255
```

https://circleci.com/gh/rubocop-hq/rubocop/25658

File: .circleci/config.yml [hunk 2]
```diff
@@ -206,7 +229,7 @@ jobs:
   # Job for downloading the Code Climate test reporter
   cc-setup:
     docker:
-      - image: circleci/ruby:2.5
+      - image: circleci/ruby:2.6
```
Comment (bquorning) [inline .circleci/config.yml]: Actually, this should probably just be `circleci/ruby` so we don’t have to update it all the time.

File: .circleci/config.yml [hunk 3]
```diff
@@ -224,7 +247,7 @@ jobs:
   # Job for merging code coverage results and sending them to Code Climate
   cc-upload-coverage:
     docker:
-      - image: circleci/ruby:2.5
+      - image: circleci/ruby:2.6
```
Comment (bquorning) [inline .circleci/config.yml]: Same here: Let’s just use `circleci/ruby`.

File: .circleci/config.yml [hunk 4]
```diff
@@ -235,13 +258,13 @@ jobs:
...
           command: |
-            ./tmp/cc-test-reporter sum-coverage tmp/codeclimate.*.json --parts 5 --output tmp/codeclimate.total.json
+            ./tmp/cc-test-reporter sum-coverage tmp/codeclimate.*.json --parts 6 --output tmp/codeclimate.total.json
             ./tmp/cc-test-reporter upload-coverage --input tmp/codeclimate.total.json
 
   # Miscellaneous tasks
   documentation-checks:
     docker:
-      - image: circleci/ruby:2.5
+      - image: circleci/ruby:2.6
```
Comment (bquorning) [inline .circleci/config.yml]: Just `circleci/ruby`.

File: .circleci/config.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (bquorning) [reply .circleci/config.yml]: The change is not related to Ruby 2.6.0 per se, but to the fact that we now get coverage data from one more build than before: ruby-2.2-spec, ruby-2.3-spec, ruby-2.4-spec, ruby-2.5-spec, ruby-2.6-spec, and jruby-9.2-spec.

File: .circleci/config.yml (hunk 2 above)
Comment (koic) [reply .circleci/config.yml]: That makes sense! I updated this PR.

**Comment by bbatsov** [comment]
👍

**Comment by koic** [comment]
I added Ruby 2.6 matrix to required status checks to pass before merging on this GitHub repo.

**Comment by jaredbeck** [comment]
I just installed 0.62 now, and seems to work well with ruby 2.6 on my project. Thanks @koic and @bbatsov! And thanks whoever does the parser!

**Pull request merged**